packages/react-reconciler/src/ReactFiberBeginWork.old.js

```diff
@@ -703,17 +715,15 @@ function updateOffscreenComponent(
       // This is the second render. The surrounding visible content has already
       // committed. Now we resume rendering the hidden tree.
 
-      if (enableCache && prevState !== null) {
+      if (enableCache && current !== null) {
         // If the render that spawned this one accessed the cache pool, resume
         // using the same cache. Unless the parent changed, since that means
         // there was a refresh.
-        const prevCachePool = prevState.cachePool;
-        if (prevCachePool !== null) {
-          spawnedCachePool = restoreSpawnedCachePool(
-            workInProgress,
-            prevCachePool,
-          );
-        }
+        const prevCachePool = prevState !== null ? prevState.cachePool : null;
+        spawnedCachePool = restoreSpawnedCachePool(
```

Comment:
Let's make move the `let spawnedCachePool` declaration into this block? We no longer reference it beyond the lifetime of this block.